Gems/Atom/Tools/MaterialCanvas/Assets/MaterialCanvas/TestData/ddx.materialcanvasnode.azasset

```diff
@@ -0,0 +1,57 @@
+{
+    "Type": "JsonSerialization",
+    "Version": 1,
+    "ClassName": "DynamicNodeConfig",
+    "ClassData": {
+        "category": "Math Functions",
+        "title": "ddx",
+        "inputSlots": [
+            {
+                "name": "inValue",
+                "displayName": "Value",
+                "description": "Value",
+                "supportedDataTypes": [
+                    "float",
+                    "float2",
+                    "float3",
+                    "float4",
+                    "color"
+                ],
+                "defaultValue": {
+                    "$type": "Vector4",
+                    "Value": [
+                        0.0,
+                        0.0,
+                        0.0,
+                        0.0
+                    ]
+                },
+                "settings": {
+                    "instructions": [ "SLOTTYPE NODEID_SLOTNAME = SLOTVALUE;" ]
+                }
+            }
+        ],
+        "outputSlots": [
+            {
+                "name": "outValue",
+                "displayName": "Value",
+                "description": "Value",
+                "supportedDataTypes": [
+                    "float4"
+                ],
+                "defaultValue": {
+                    "$type": "Vector4",
```

Comment:
Can we make the output type automatically match the input type?

Reply:
Yes. We'll do that in that in code. Still need to specify the type for the default value